VmAgent.Core/Interfaces/SystemOperations.cs

```diff
@@ -261,5 +300,50 @@ public void SetUnixFilePermissions(string filePath, int permissions)
             var unixFileInfo = new UnixFileInfo(filePath);
             unixFileInfo.FileAccessPermissions = (FileAccessPermissions)permissions;
         }
+
+        public void SetUnixOwnerIfNeeded(string path, bool applyToAllContents = false)
+        {
+            if (_vmConfiguration != null && _vmConfiguration.RunContainersInUsermode)
+            {
+                _logger.LogInformation($"Setting unix owner for {path}");
+
+                _fileSystemOperations.SetUnixOwner(path, User);
+
+                if (_fileSystemOperations.IsDirectory(path) && applyToAllContents)
+                {
+                    DirectoryInfo dirInfo = new DirectoryInfo(path);
+                    IEnumerable<FileSystemInfo> childItems = _fileSystemOperations.GetFileSystemInfos(dirInfo, true);
+
+                    foreach (FileSystemInfo child in childItems)
+                    {
+                        _fileSystemOperations.SetUnixOwner(child.FullName, User);
+                    }
+                }
+            }
+            else
+            {
+                _logger.LogInformation("Unix file setting not needed");
+            }
+        }
+
+        public void CreateDirectoryAndParents(DirectoryInfo directory)
```

Comment:
Are we doing it this way because we want every folder in the hierarchy to be owned by the user? Otherwise, we can just do Directory.CreateDirectory right? It is supposed to create all the necessary folders in the path. And then you can just chown the leaf.